conformance/tests/httproute-redirect-port.go

```diff
@@ -38,70 +39,172 @@ var HTTPRouteRedirectPort = suite.ConformanceTest{
 	Features:    []suite.SupportedFeature{suite.SupportHTTPRoutePortRedirect},
 	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
 		ns := "gateway-conformance-infra"
-		routeNN := types.NamespacedName{Name: "redirect-port", Namespace: ns}
+
 		gwNN := types.NamespacedName{Name: "same-namespace", Namespace: ns}
-		gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+		routeNN := types.NamespacedName{Name: "http-route-for-listener-on-port-80", Namespace: ns}
+		gwAddr80 := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
 		kubernetes.HTTPRouteMustHaveResolvedRefsConditionsTrue(t, suite.Client, suite.TimeoutConfig, routeNN, gwNN)
 
-		testCases := []http.ExpectedResponse{{
-			Request: http.Request{
-				Path:             "/port",
-				UnfollowRedirect: true,
-			},
-			Response: http.Response{
-				StatusCode: 302,
-			},
-			RedirectRequest: &roundtripper.RedirectRequest{
-				Port: "8083",
-			},
-			Namespace: ns,
-		}, {
-			Request: http.Request{
-				Path:             "/port-and-host",
-				UnfollowRedirect: true,
-			},
-			Response: http.Response{
-				StatusCode: 302,
+		gwNN = types.NamespacedName{Name: "gateway-with-listener-on-443", Namespace: ns}
+		routeNN = types.NamespacedName{Name: "http-route-for-listener-on-port-443", Namespace: ns}
+		gwAddr443 := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+		kubernetes.HTTPRouteMustHaveResolvedRefsConditionsTrue(t, suite.Client, suite.TimeoutConfig, routeNN, gwNN)
+
+		certNN := types.NamespacedName{Name: "tls-validity-checks-certificate", Namespace: string(ns)}
+		cPem, keyPem, err := GetTLSSecret(suite.Client, certNN)
+		if err != nil {
+			t.Fatalf("unexpected error finding TLS secret: %v", err)
+		}
+
+		testCases := []struct {
+			expectedResponse http.ExpectedResponse
```

Comment:
I'm not sure I understand this, does this mean the request is now embedded in the expected response?

Reply:
The comment may be obsolete but yes I may share that opinion.

If we're talking about the same thing, then yes I do find the existing naming of the struct `ExpectedResponse` confusing -- it does embed the request to be made (which is counterintuitive from the name)